containers/base/Dockerfile

```diff
@@ -1,252 +1,293 @@
-RUN ipython profile create default && \
-    jupyter notebook --generate-config
-
-RUN if [ -d /datalab/lib/pydatalab/.git ]; then \
-        echo "use local lib"; \
-      else \
-        git clone https://github.com/googledatalab/pydatalab.git /datalab/lib/pydatalab; \
-      fi
-RUN cd /datalab/lib/pydatalab && \
-    /tools/node/bin/npm install -g typescript && \
-    tsc --module amd --noImplicitAny --outdir datalab/notebook/static datalab/notebook/static/*.ts && \
-    tsc --module amd --noImplicitAny --outdir google/datalab/notebook/static google/datalab/notebook/static/*.ts && \
-    /tools/node/bin/npm uninstall -g typescript
-RUN cd /datalab/lib/pydatalab && \
-    pip install --upgrade-strategy only-if-needed --no-cache-dir . && \
-    pip install --upgrade-strategy only-if-needed /datalab/lib/pydatalab/solutionbox/image_classification/. && \
-    pip install --upgrade-strategy only-if-needed /datalab/lib/pydatalab/solutionbox/structured_data/. && \
-    pip3 install --upgrade-strategy only-if-needed --no-cache-dir . && \
-    pip3 install --upgrade-strategy only-if-needed /datalab/lib/pydatalab/solutionbox/image_classification/. && \
-    pip3 install --upgrade-strategy only-if-needed /datalab/lib/pydatalab/solutionbox/structured_data/. && \
-    jupyter nbextension install --py datalab.notebook && \
-    jupyter nbextension install --py google.datalab.notebook && \
-    jupyter nbextension enable --py widgetsnbextension && \
-    rm datalab/notebook/static/*.js google/datalab/notebook/static/*.js && \
-    mkdir -p /datalab/nbconvert && \
-    cp -R /usr/local/share/jupyter/nbextensions/gcpdatalab/* /datalab/nbconvert && \
-    ln -s /usr/local/lib/python2.7/dist-packages/notebook/static/custom/custom.css /datalab/nbconvert/custom.css && \
-    cd /
-
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# We use different base images for GPU vs CPU Dockerfiles, so we expect
+# that the appropriate image is pulled and tagged locally.
+# CPU should use ubuntu:16.04
+# and GPU uses nvidia/cuda:8.0-cudnn5-devel-ubuntu16.04
+FROM datalab-external-base-image
+MAINTAINER Google Cloud DataLab
+
+# Container configuration
+EXPOSE 8080
+
+# Path configuration
+ENV PATH $PATH:/tools/node/bin:/tools/google-cloud-sdk/bin
+ENV PYTHONPATH /env/python
+
+RUN apt-get update -y && apt-get install -y --no-install-recommends apt-utils

```

Comment:
You can move `apt-utils` down into the next `RUN` layer to avoid a large docker image size.

Reply:
Also, why is this package needed?

Reply:
I think that `apt-utils` is not actually needed by inspecting for example Rocker's various dockerfile compositions ([here](https://github.com/rocker-org/rocker/search?q=utils&type=Code&utf8=%E2%9C%93)).